File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> mp(26, 0);
        for (int i = 0; i < s.size(); i++) {
            mp[s[i]-'a']++;
        }
        for (int i = 0; i < t.size(); i++) {
            mp[t[i]-'a']--;
        }
        for (int i = 0; i < 26; i++) {
            if (mp[i] != 0) {
                return false;
            }
        }
        return true;
    }
};

File: Data Structures & Algorithms/partition-labels/submission-0.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        mp = [0] * 26
        for i in range(len(s)):
            mp[ord(s[i]) - ord('a')] = i

        res = []
        l = r = 0
        for i in range(len(s)):
            if mp[ord(s[i]) - ord('a')] == r and i == r:
                res.append(i - l + 1)
                l = r = i + 1
            else:
                r = max(r, mp[ord(s[i]) - ord('a')])
        return res